packages/protocol/test-sol/integration/Integration.t.sol

```diff
@@ -1,19 +1,81 @@
+
 import "@celo-contracts/common/interfaces/IRegistry.sol";
+import "@celo-contracts/common/interfaces/IProxy.sol";
 
 contract IntegrationTest is Test {
   address constant registryAddress = address(0x000000000000000000000000000000000000ce10);
-  address account1 = actor("account1");
-  address account2 = actor("account2");
   IRegistry registry = IRegistry(registryAddress);
 
   function setUp() public {}
+}
+
+contract RegistryIntegrationTest is IntegrationTest, Utils, Constants {
+  IProxy proxy;
+
+  function test_shouldHaveAddressInRegistry() public view {
+    for (uint256 i = 0; i < contractsInRegistry.length; i++) {
+      string memory contractName = contractsInRegistry[i];
+      address contractAddress = registry.getAddressFor(keccak256(abi.encodePacked(contractName)));
+      console2.log(contractName, "address in Registry is: ", contractAddress);
+      assert(contractAddress != address(0));
+    }
+  }
+
+  function test_shouldHaveCorrectBytecode() public {
+    for (uint256 i = 0; i < contractsInRegistry.length; i++) {
+      // Read name from list of core contracts
+      string memory contractName = contractsInRegistry[i];
+      console2.log("Checking bytecode of:", contractName);
+
+      // Converting contract names to hashes for comparison
+      bytes32 hashContractName = keccak256(abi.encodePacked(contractName));
+      bytes32 hashAccount = keccak256(abi.encodePacked("Accounts"));
+      bytes32 hashElection = keccak256(abi.encodePacked("Election"));
+      bytes32 hashEscrow = keccak256(abi.encodePacked("Escrow"));
+      bytes32 hashFederatedAttestations = keccak256(abi.encodePacked("FederatedAttestations"));
+      bytes32 hashGovernance = keccak256(abi.encodePacked("Governance"));
+      bytes32 hashSortedOracles = keccak256(abi.encodePacked("SortedOracles"));
+      bytes32 hashValidators = keccak256(abi.encodePacked("Validators"));
+
+      // Skipping test for contracts that depend on linked libraries
+      // This is a known limitation in Foundry at the moment:
+      // Source: https://github.com/foundry-rs/foundry/issues/6120
+      if (
+        hashContractName != hashAccount &&
+        hashContractName != hashElection &&
+        hashContractName != hashEscrow &&
+        hashContractName != hashFederatedAttestations &&
+        hashContractName != hashGovernance &&
+        hashContractName != hashSortedOracles &&
+        hashContractName != hashValidators
+      ) {
+        // Get proxy address registered in the Registry
+        address proxyAddress = registry.getAddressForStringOrDie(contractName);
+        proxy = IProxy(address(uint160(proxyAddress)));
+
+        // Get implementation address
+        address implementationAddress = proxy._getImplementation();
+
+        // Get bytecode from deployed contract
+        bytes memory actualBytecodeWithMetadata = getCodeAt(implementationAddress);
+        bytes memory actualBytecode = removeMetadataFromBytecode(actualBytecodeWithMetadata);
```

Comment:
Can we update the name to reflect that this code is taken from devchain (so it is absolutely clear) eg actualBytecodeOnDevchain

Reply:
Yes, that's done in https://github.com/celo-org/celo-monorepo/pull/11002/commits/b9ce12943d5ae699c4867fd61fb8b1ff8bb210ed